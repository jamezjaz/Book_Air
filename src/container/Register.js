import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import registration from '../apiRequests/regRequest';

const Register = props => {
  const user = useSelector(state => state.user);
  const dispatch = useDispatch();
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleChange = event => {
    if (event.target.name === 'name') {
      setName(event.target.value);
    } else if (event.target.name === 'email') {
      setEmail(event.target.value);
    } else {
      setPassword(event.target.value);
    }
    // console.log(event.target);
  };

  const handleRegister = () => {
    const { history } = props;
    history.push('/airlinesList');
  };

  const handleSubmit = event => {
    event.preventDefault();
    console.log('Initial User', user);
    dispatch(registration({
      name,
      email,
      password,
    }));
    handleRegister();
  };

  return (
    <div>
      <form>
        <label htmlFor="name">
          Name:
          <input type="text" name="name" id="name" placeholder="Enter your name" onChange={handleChange} value={name.name} />
        </label>
        <br />
        <label htmlFor="email">
          Email:
          <input type="text" name="email" id="email" placeholder="Enter your email" onChange={handleChange} value={email.email} />
        </label>
        <br />
        <label htmlFor="password">
          Password:
          <input type="password" name="password" id="password" placeholder="Enter your password" onChange={handleChange} value={password.password} />
        </label>
        <br />
        <button type="submit" onClick={handleSubmit}>Register</button>
      </form>
    </div>
  );
};

Register.propTypes = {
  history: PropTypes.instanceOf(Object).isRequired,
};

export default Register;
