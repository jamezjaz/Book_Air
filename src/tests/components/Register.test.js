import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Register from '../../container/Register';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Register', () => {
  test('matches Register snapshot', () => {
    const tree = renderer
      .create(
        (<Provider store={store}>
          <Router>
            <Register />
          </Router>
        </Provider>),
        )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders Register component', () => {
    const div = document.createElement('div');
    const { queryAllByTestId } = render(
      (<Provider store={store}>
        <Router>
          <Route path="/register" component={Register} />
        </Router>
      </Provider>),
      );
    const container = queryAllByTestId('container');
    expect(container).toBeTruthy();
  });

  test('it should have a Register form', () => {
    const div = document.createElement('div');
    const { queryAllByTestId } = render(
      (<Provider store={store}>
        <Router>
          <Route path="/register" component={Register} />
        </Router>
      </Provider>),
      );
    const form = queryAllByTestId('form');
    expect(form).toBeTruthy();
  });
});
