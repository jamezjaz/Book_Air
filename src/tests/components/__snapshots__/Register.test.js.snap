// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register matches Register snapshot 1`] = `
<div
  className="container-fluid"
  data-testId="container"
>
  <a
    className="d-flex justify-content-end"
    href="/"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      Home
    </button>
  </a>
  <form
    data-testId="form"
    onSubmit={[Function]}
  >
    <label
      className="text-left"
      htmlFor="name"
    >
      Name:
      <br />
      <input
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Enter your name"
        type="text"
      />
    </label>
    <br />
    <label
      className="text-left"
      htmlFor="username"
    >
      Username:
      <br />
      <input
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Enter your username"
        type="text"
      />
    </label>
    <br />
    <label
      className="text-left"
      htmlFor="email"
    >
      Email:
      <br />
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="text"
      />
    </label>
    <br />
    <label
      className="text-left"
      htmlFor="password"
    >
      Password:
      <br />
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
      />
    </label>
    <br />
    <button
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
    <div>
      <span>
        Already have an account? 
      </span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
    
  </form>
</div>
`;
